bftengine/src/bcstatetransfer/Messages.hpp

```diff
@@ -53,56 +55,110 @@ struct AskForCheckpointSummariesMsg : public BCStateTranBaseMsg {
 };
 
 struct CheckpointSummaryMsg : public BCStateTranBaseMsg {
-  CheckpointSummaryMsg() {
-    // NOLINTNEXTLINE(bugprone-undefined-memory-manipulation)
-    memset(this, 0, sizeof(CheckpointSummaryMsg));
-    type = MsgType::CheckpointsSummary;
+  CheckpointSummaryMsg() = delete;
+
+  static CheckpointSummaryMsg* create(size_t rvbDataSize) {
+    size_t totalSize = sizeof(CheckpointSummaryMsg) + rvbDataSize - 1;
```

Comment:
Maybe rename to totalByteSize (and the parameter as well, as it is not clear what size unit it uses)?
What is the minus 1 for?
Why not change the data member to a unique ptr of a byte array?

Reply:
- totalByteSize -> OK thanks, I'll change it.
- What is the minus 1 for -> substruct data[1] (dynamic array size)
- Having a vector gives many multiple advantages, for example I do not need to check the size, is it empty and many other things )I get many simple API functions which simplify the code) In general, vector is a one of the best solutions after a unique pointer to keep a buffer - while it also manages its own memory.